ap_src/ap_app/utils/switch_permissions.py

```diff
@@ -0,0 +1,68 @@
+# Models
+from ..models import (UserProfile)
+from django.contrib.auth.models import User
+
+# Utils
+from teams_utils import get_users_sharing_teams
+from user_profile import get_user_id_from_username
+from user_profile import get_userprofile_id_from_user_id
+
+# this check should be activated when the user leaves a team
+def check_for_lingering_switch_perms(username): # stops users from having switch perms when they don't share any teams
+    user_id = get_user_id_from_username(username)
+    userprofile_id = get_userprofile_id_from_user_id(user_id)
+
+    usernames_given_permissions, userprofile_usernames_who_give_permissions = get_associated_permissions(username, userprofile_id, user_id)
+
+    user = User.objects.get(id=user_id)
+    users_sharing_teams = get_users_sharing_teams(username, user)
+    
+    process_user_usernames(username, usernames_given_permissions, users_sharing_teams)
+    process_userprofile_usernames(username, userprofile_usernames_who_give_permissions, users_sharing_teams)
+    
+    """
```

Comment:
I'm not sure what this multiline comment is for, you can just remove it.

Reply:
The comment was some pseudocode to show the basic structure of how the algorithm works mostly for my benefit but it has since changed slightly so it will be removed.

Reply:
Fixed with 717b45d